graphql/batch_executor.go

```diff
@@ -10,6 +10,14 @@ import (
 	"github.com/samsarahq/thunder/reactive"
 )
 
+type executeFunction func(ctx context.Context) error
+
+type ExecutionUnit struct {
```

Comment:
I think a comment would be helpful here.  What is the difference between a work unit and an execution unit?  When would I need to use the former vs the latter?